broker/src/main/java/io/camunda/zeebe/broker/partitioning/PartitionManagerImpl.java

```diff
@@ -123,64 +127,68 @@ public CompletableFuture<Void> start() {
 
     healthCheckService.registerBootstrapPartitions(memberPartitions);
     for (final var partitionMetadata : memberPartitions) {
-      final var id = partitionMetadata.id().id();
-      partitionStartup
-          .bootstrap(partitionMetadata)
-          .whenComplete(
-              (startedPartition, throwable) -> {
-                if (throwable != null) {
-                  LOGGER.error("Failed to start partition {}", id, throwable);
-                  onHealthChanged(id, HealthStatus.DEAD);
-                } else {
-                  LOGGER.info("Started partition {}", id);
-                  final var zeebePartition = startedPartition.zeebePartition();
-                  final var raftPartition = startedPartition.raftPartition();
-
-                  zeebePartition.addFailureListener(
-                      new PartitionHealthBroadcaster(id, this::onHealthChanged));
-                  diskSpaceUsageMonitor.addDiskUsageListener(zeebePartition);
-                  adminAccess.put(id, zeebePartition.getAdminAccess());
-                  zeebePartitions.add(zeebePartition);
-                  raftPartitions.add(raftPartition);
-                }
-              });
+      bootstrapPartition(partitionMetadata);
     }
-    return CompletableFuture.completedFuture(null);
-  }
-
-  public CompletableFuture<Void> stop() {
-    return stopZeebePartitions()
-        .thenCompose((ignored) -> stopRaftPartitions())
-        .whenComplete(
-            (ok, error) -> {
-              if (error != null) {
-                LOGGER.error(error.getMessage(), error);
-              }
-              raftPartitions.clear();
-              zeebePartitions.clear();
-              adminAccess.clear();
-              topologyManager.close();
-            });
   }
 
-  private CompletableFuture<Void> stopZeebePartitions() {
-    final var futures =
-        zeebePartitions.stream()
-            .map(partition -> CompletableFuture.runAsync(() -> stopPartition(partition)))
-            .toArray(CompletableFuture[]::new);
-
-    return CompletableFuture.allOf(futures);
-  }
-
-  private CompletableFuture<Void> stopRaftPartitions() {
-    return CompletableFuture.allOf(
-        raftPartitions.stream().map(RaftPartition::close).toArray(CompletableFuture[]::new));
+  private void bootstrapPartition(final PartitionMetadata partitionMetadata) {
+    final var id = partitionMetadata.id().id();
+    final var context =
+        new PartitionStartupContext(
+            actorSchedulingService,
+            concurrencyControl,
+            managementService,
+            partitionMetadata,
+            raftPartitionFactory,
+            zeebePartitionFactory,
+            brokerCfg);
+    final var partition = Partition.bootstrapping(context);
+    partitions.put(id, partition);
+
+    concurrencyControl.run(
```

Comment:
:question: Why do you have to wrap it in `concurrencyControl::run`? `runOnCompletion` ensures that the consumer is run on this actor..

Reply:
`runOnCompletion` can only be called from an actor :upside_down_face:

Reply:
But isn't it running in an actor already? If testing is the problem, you can use `onComplete`.

Reply:
Or, if you are sure that it is not running in the actor, you can pass the executor to `onComplete`.

Reply:
You are right, this should already be called from the correct actor. I guess I observed this failing while I was still ~~breaking~~ refactoring things :+1: